keps/sig-scheduling/5278-nominated-node-name-for-expectation/README.md

```diff
@@ -0,0 +1,871 @@
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Use `NominatedNodeName` to express an pod placement, expected by the scheduler or expected by other components.
+
+The scheduler puts `NominatedNodeName` at the beginning of binding cycles to show an expected pod placement to other components.
+And, also other components can put `NominatedNodeName` on pending pods to indicate the pod is prefered to be scheduled on a specific node.
+
+## Motivation
+
+### External components need to know the pod is going to be bound 
+
+The scheduler reserves the place for the pod when the pod is entering the binding cycle.
+This reservation is internally implemented in the scheduler's cache, and is not visible to other components.
+
+The specific problem is, as shown at [#125491](https://github.com/kubernetes/kubernetes/issues/125491),
+if the binding cycle takes time before binding pods to nodes (e.g., PreBind takes time to handle volumes)
+the cluster autoscaler cannot understand the pod is going to be bound there,
+misunderstands the node is low-utilized (because the scheduler keeps the place of the pod), and deletes the node. 
+
+We can expose those internal reservations with `NominatedNodeName` so that external components can take a more appropriate action
+based on the expected pod placement.
+
+### External components want to specify a preferred pod placement
+
+The cluster autoscaler or Kueue internally calculates the pod placement,
+and create new nodes or un-gate pods based on the calculation result. 
+
+So, they know where those pods are likely going to be scheduled.
+
+By specifing their expectation on `NominatedNodeName`, the scheduler can first check whether the pod can go to the nominated node,
+speeding up the filter phase.
```

Comment:
@mimowo TBH, I want to avoid pre-evaluation of Pod scheduling directives on Kueue side since that means Kueue needs to evaluate all Pod scheduling directives the same as kube-scheduler.

Reply:
I would defiently prefer to avoid duplicating the code, but maybe it is possible for kube-scheduler to expose helpers which could be used by Kueue.